### PR TITLE
fitboot: boot a bzImage from a FIT

### PR DESCRIPTION
**Pull request by rminnich**
Read a fit image file specified by the first arg.
boot a bzimage specified by the node name in the second arg.

Signed-off-by: Ronald G Minnich <rminnich@gmail.com>

### CODE REVIEW

File: cmds/boot/fitboot/main.go [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+	dryRun    = flag.Bool("dryrun", true, "Do not actually kexec into the boot config")
+	debug     = flag.Bool("d", true, "Print debug output")
+	rootfs    = flag.String("r", "", "Root file system name")
+	cmdline   = flag.String("c", "earlyprintk=ttyS0,115200,keep console=ttyS0", "command line")
+	initramfs = flag.String("i", "", "InitRAMFS node name -- default none")
+)
+
+var v = func(string, ...interface{}) {}
+
+func main() {
```
Comment (rjoleary) [inline cmds/boot/fitboot/main.go:25]: Couldn't this be part of kexec? Why a separate command?
The magic for fit is 0xd00dfeed.

File: cmds/boot/fitboot/main.go [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+	flag.Parse()
+	if *debug {
+		v = log.Printf
+	}
+	if len(flag.Args()) < 2 {
+		log.Fatal("Usage: fitboot <file> <node name>")
```
Comment (rjoleary) [inline cmds/boot/fitboot/main.go:31]: Could you somehow make "<node name>" default to the first kernel?

File: cmds/boot/fitboot/main.go (hunk 1 above)
Comment (rminnich) [reply cmds/boot/fitboot/main.go:25]: I'd rather it be separate. It's a really special case, currently used by only one user, and I hate to bloat kexec up with it.

File: cmds/boot/fitboot/main.go (hunk 2 above)
Comment (rminnich) [reply cmds/boot/fitboot/main.go:31]: Next round if that's ok :-)

but not like that. I expect we won't need to name the kernel.

File: go.mod [hunk 3]
```diff
@@ -28,6 +28,7 @@ require (
 	github.com/mdlayher/raw v0.0.0-20191009151244-50f2db8cc065 // indirect
 	github.com/rck/unit v0.0.3
 	github.com/rekby/gpt v0.0.0-20200219180433-a930afbc6edc
+	github.com/rminnich/go-lzo v0.0.0-20201110235843-e5cfc6a1d9af
```
Comment (rjoleary) [inline go.mod:31]: Why is this lzo stuff added?

**Review by rjoleary — APPROVED** [review]

File: go.mod (hunk 3 above)
Comment (rminnich) [reply go.mod:31]: no idea. It used to be used. Maybe we need to run go mod tidy on this again?

**Comment by rminnich** [comment]
obsoleted by #1887

**Pull request closed without merging**